neo/Network/P2P/TaskManager.cs

```diff
@@ -139,8 +176,8 @@ private void OnTimer()
                 foreach (var task in session.Tasks.ToArray())
                     if (DateTime.UtcNow - task.Value > TaskTimeout)
                     {
-                        globalTasks.Remove(task.Key);
-                        session.Tasks.Remove(task.Key);
+                        if (session.Tasks.Remove(task.Key) && task.Key != UInt256.Zero)
```

Comment:
Why check `task.Key != UInt256.Zero`?

Reply:
Because tasks has 0 are header tasks and will never be in globalTasks. It is an optimization.

Reply:
Since it is infrequently 0 it is probably a poor optimization though; I’ll Remove it.